### PR TITLE
1560 - Apply area and address specific rental property rules when creating applicant in internal portal

### PR DESCRIPTION
**Pull request by momentiris**
- Adds route for getting a tenant

A tenant is something I think is different than a contact. A tenant should be someone with at least one housing contract.

### CODE REVIEW

File: src/services/lease-service/get-tenant.ts [hunk 1]
```diff
@@ -18,11 +18,17 @@ type GetTenantError =
   | 'housing-contracts-not-found'
   | 'get-lease-property-info'
 
+// TODO: Tenant should somehow represent that is definitely has at least one
```
Comment (driatic) [inline src/services/lease-service/get-tenant.ts:21]: I agree that this is a good definition of a tenant!

File: src/services/lease-service/routes/applicants.ts [hunk 2]
```diff
@@ -315,9 +315,21 @@ export const routes = (router: KoaRouter) => {
    *             schema:
    *               type: object
    *               properties:
+   *                 applicationType: string
```
Comment (driatic) [inline src/services/lease-service/routes/applicants.ts:318]: 👍

**Review by driatic — APPROVED** [review]
Nice work! Merge away

**Review by driatic — CHANGES_REQUESTED** [review]
Forgot about leaving feedback on the response formats... I think we should implement tha changes made by Johannes as per: https://github.com/Bostads-AB-Mimer/onecore-leasing/commit/b3c62362a5e947c6dc38d396c5ef530d68428fc0

File: src/services/lease-service/get-tenant.ts [hunk 3]
```diff
@@ -18,11 +18,17 @@ type GetTenantError =
...
+// active housing contract. Otherwise, it wouldn't be a tenant, right?
+
+type NonEmptyArray<T> = [T, ...T[]]
+
 export type Tenant = Omit<Contact, 'leases' | 'isTenant'> & {
```
Comment (driatic) [inline src/services/lease-service/get-tenant.ts:26]: Can we export this type to onecore-types?

File: src/services/lease-service/routes/contacts.ts [hunk 4]
```diff
@@ -171,6 +172,55 @@ export const routes = (router: KoaRouter) => {
...
+   *               properties:
+   *                 data:
+   *                   type: object
+   *                   description: The tenant data.
+   *       404:
+   *         description: Not found.
+   *       500:
+   *         description: Internal server error. Failed to retrieve Tenant information.
+   */
+  router.get('(.*)/tenant/contactCode/:contactCode', async (ctx) => {
```
Comment (driatic) [inline src/services/lease-service/routes/contacts.ts:205]: `tenants` to align with DIGG-standard

Comment (momentiris) [reply src/services/lease-service/routes/contacts.ts:205]: Is that to add the `metadata` property to the response?

Comment (momentiris) [reply src/services/lease-service/routes/contacts.ts]: https://github.com/Bostads-AB-Mimer/onecore-leasing/pull/109/commits/bbb24996e46cc910afa24c357bd70e477f808ae1

File: src/services/lease-service/get-tenant.ts (hunk 3 above)
Comment (momentiris) [reply src/services/lease-service/get-tenant.ts:27]: Will apply after this is merged:
https://github.com/Bostads-AB-Mimer/onecore-types/pull/87

Comment (momentiris) [reply src/services/lease-service/get-tenant.ts]: https://github.com/Bostads-AB-Mimer/onecore-leasing/pull/109/commits/9165a8b7b33eaed323470903446f2da1c87921ef

**Review by driatic — APPROVED** [review]
🔥

**Pull request merged**